python/ray/train/_internal/backend_executor.py

```diff
@@ -509,12 +518,24 @@ def get_next():
 
         return results
 
-    def _set_checkpoint_uri(self, uri: str):
+    def _set_checkpoint_id(self, checkpoint_id: int):
```

Comment:
Shall we use the term "number" instead of id, to hint that they are sequentially assigned and not random?

Reply:
How about "checkpoint_index"?

Reply:
Sure.